ffi/diplomat/src/casemap.rs

```diff
@@ -0,0 +1,166 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+#[diplomat::bridge]
+pub mod ffi {
+    use crate::{
+        errors::ffi::ICU4XError, locale::ffi::ICU4XLocale, provider::ffi::ICU4XDataProvider,
+    };
+    use alloc::boxed::Box;
+    use diplomat_runtime::DiplomatWriteable;
+    use icu_casemap::CaseMapper;
+    use writeable::Writeable;
+
+    #[diplomat::opaque]
+    #[diplomat::rust_link(icu::casemap::CaseMapper, Struct)]
+    pub struct ICU4XCaseMapper(pub CaseMapper);
+
+    impl ICU4XCaseMapper {
+        /// Construct a new ICU4XCaseMapper instance for NFC
+        #[diplomat::rust_link(icu::casemap::CaseMapper::new, FnInStruct)]
+        pub fn create(provider: &ICU4XDataProvider) -> Result<Box<ICU4XCaseMapper>, ICU4XError> {
+            Ok(Box::new(ICU4XCaseMapper(call_constructor!(
+                CaseMapper::new [r => Ok(r)],
```

Comment:
thought: should the compiled data constructor return a `CaseMapperBorrowed` type?

Reply:
seems unnecessary to me